### PR TITLE
Init skeleton for dask backend

### PR DESCRIPTION
**Pull request by gerrymanoim**
Part one of four in breaking up https://github.com/ibis-project/ibis/pull/2400 into more manageable pieces. 

Goal in this PR is to create the basic skeleton and get CI running. I've also added conftest pieces that didn't depend on anything else.

### CODE REVIEW

File: ci/deps/dask.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+dask
```
Comment (jreback) [inline ci/deps/dask.yml:1]: do we need to list distributed at all? (even if for testing)?

**Review by jreback — CHANGES_REQUESTED** [review]
can you add a trivial test that just does the import (so we know its running)

ping on green

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     name: Tests pandas / files
     runs-on: ${{ matrix.os }}
     env:
-      BACKENDS: "pandas csv parquet hdf5"
+      BACKENDS: "pandas dask csv parquet hdf5"
```
Comment (jreback) [inline .github/workflows/main.yml:14]: let's add this as a separate GH (basically copy the pandas section to a new Tests_dask)

**Comment by jreback** [comment]
excellent, ping on green. on the TODO list will need to add docs / whatsnew note

File: ci/deps/dask.yml (hunk 1 above)
Comment (gerrymanoim) [reply ci/deps/dask.yml:1]: Hmmm - good question. I wasn't using it for any of my tests previously (just relying on the default scheduler), but I think it is worth adding in the future. 

Would the expectation be that users pass in the `dask.distributed.Client` to the ibis client?

**Comment by gerrymanoim** [comment]
@jreback - This will be "green" with the exception of the pandas test that failed because of a slow import. 

```
    @pytest.mark.skipif(
        bool(os.environ.get("AZURECI")),
        reason="Testing import time on CI is flaky due to machine variance",
    )
    def test_import_time():
        pb = pytest.importorskip("plumbum")
        lines = ["from timeit import timeit", "print(timeit('import ibis'))"]
        delta = float(pb.cmd.python["-c", "; ".join(lines)]().strip())
>       assert delta < 2.0
E       assert 2.0281344 < 2.0
```

Do you want me to bump that?

**Comment by jreback** [comment]
> @jreback - This will be "green" with the exception of the pandas test that failed because of a slow import.
> 
> ```
>     @pytest.mark.skipif(
>         bool(os.environ.get("AZURECI")),
>         reason="Testing import time on CI is flaky due to machine variance",
>     )
>     def test_import_time():
>         pb = pytest.importorskip("plumbum")
>         lines = ["from timeit import timeit", "print(timeit('import ibis'))"]
>         delta = float(pb.cmd.python["-c", "; ".join(lines)]().strip())
> >       assert delta < 2.0
> E       assert 2.0281344 < 2.0
> ```
> 
> Do you want me to bump that?
> 
> RE: docs/whatsnew - my plan was to add that as the dask backend reached some MVP status.

the import test is fine (we can adjust it later) we just don't want it to be *much* greater.

> RE: docs/whatsnew - my plan was to add that as the dask backend reached some MVP status.

yep exactly. pls open a new issue for the dask backend (so you can edit), and put the TODO's in the issue as checkboxes.

File: ibis/backends/dask/tests/test_import.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Trivial test to make sure CI runs and package is importable
```
Comment (jreback) [inline ibis/backends/dask/tests/test_import.py:1]: can you show me on the CI where this ran?

Comment (gerrymanoim) [reply ibis/backends/dask/tests/test_import.py:1]: Oh great catch. Looks like `run_tests.sh` still expects the directory structure to be `ibis/$BACKEND/tests` (see https://github.com/ibis-project/ibis/blob/master/ci/run_tests.sh#L8), where I think given the new directory structure this should be ` ibis/backends/$BACKEND/tests`. 

This is a problem in the regular tests as well: https://github.com/ibis-project/ibis/pull/2536/checks?check_run_id=1433682341#step:4:6 if I understand this correctly.

File: ci/deps/dask.yml (hunk 1 above)
Comment (jreback) [reply ci/deps/dask.yml:1]: yeah we would want to pass this to the ibis client i think.

File: ibis/backends/dask/tests/test_import.py (hunk 3 above)
Comment (jreback) [reply ibis/backends/dask/tests/test_import.py:1]: oh! yes pls split out to a separate PR if you would.

Comment (gerrymanoim) [reply ibis/backends/dask/tests/test_import.py:1]: https://github.com/ibis-project/ibis/pull/2538

File: ci/deps/dask.yml (hunk 1 above)
Comment (gerrymanoim) [reply ci/deps/dask.yml:1]: I've added this to the TODO list in https://github.com/ibis-project/ibis/issues/2537. I'll update the requirements for distributed when I add that.

File: ibis/backends/dask/tests/test_import.py (hunk 3 above)
Comment (gerrymanoim) [reply ibis/backends/dask/tests/test_import.py:1]: I've confirmed CI is now running: 

```
ibis/backends/dask/tests/test_import.py .                                [100%]
```

**Comment by gerrymanoim** [comment]
@jreback - This is now "green" with the exception of the pandas import timing check and the building of the ibis container timeout.

**Review by jreback — APPROVED** [review]
excellent, thanks @gerrymanoim

**Pull request merged**